pkg/cloud/network.go

```diff
@@ -89,21 +89,33 @@ func (c *client) GetOrCreateNetwork(csCluster *infrav1.CloudStackCluster) (retEr
 }
 
 func (c *client) ResolvePublicIPDetails(csCluster *infrav1.CloudStackCluster) (*cloudstack.PublicIpAddress, error) {
+	ip := csCluster.Spec.ControlPlaneEndpoint.Host
+
 	p := c.cs.Address.NewListPublicIpAddressesParams()
 	p.SetAllocatedonly(false)
 	setIfNotEmpty(csCluster.Spec.Account, p.SetAccount)
 	setIfNotEmpty(csCluster.Status.DomainID, p.SetDomainid)
-	if ip := csCluster.Spec.ControlPlaneEndpoint.Host; ip != "" {
+	if ip != "" {
 		p.SetIpaddress(ip)
 	}
 	publicAddresses, err := c.cs.Address.ListPublicIpAddresses(p)
+
 	if err != nil {
 		return nil, err
-	} else if publicAddresses.Count > 0 {
+	} else if ip != "" && publicAddresses.Count == 1 { // Endpoint specified and IP found.
```

Comment:
This is a fix that actually checks for public IP address availability.

Previously we would just the first IP address returned.